recipe/meta.yaml

```diff
@@ -1,90 +1,57 @@
+  string: mps_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}                                                 # [gpu_variant == "metal"]
+  missing_dso_whitelist:                                                                         # [s390x or (gpu_variant == "cuda-11") or (gpu_variant == "cuda-12")]
+    - "**/libcuda.so*"                                                                           # [(gpu_variant == "cuda-11") or (gpu_variant == "cuda-12")]
+    - '$RPATH/ld64.so.1'  # [s390x]
+  
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }}                        # [(cuda_compiler_version or "").startswith("12")]
+    - {{ compiler('cuda') }}                              # [gpu_variant == "cuda-12"]
     - cmake
-    - git
-    - ninja
+    - ninja-base
     - pkgconfig
+    - patch     # [unix]
+    - m2-patch  # [win]
   host:
-    - cudatoolkit      {{ cuda_compiler_version }}  # [cuda_compiler_version != "None" and not (cuda_compiler_version or "").startswith("12")]
-    - cuda-version     {{ cuda_compiler_version }}  # [cuda_compiler_version != "None"]
-    - cuda-cudart-dev  {{ cuda_compiler_version }}  # [(cuda_compiler_version or "").startswith("12")]
-    - libcublas-dev    {{ cuda_compiler_version }}  # [(cuda_compiler_version or "").startswith("12")]
-
-    - blas-devel * *{{ blas_impl }}                 # [not osx and cuda_compiler_version == "None" and blas_impl == "mkl"]
-    - mkl-devel {{ mkl }}                           # [not osx and cuda_compiler_version == "None" and blas_impl == "mkl"]
+    - cudatoolkit      {{ cuda_compiler_version }}        # [gpu_variant == "cuda-11"]
+    - cuda-version     {{ cuda_compiler_version }}        # [(gpu_variant == "cuda-11") or (gpu_variant == "cuda-12")]
+    - cuda-cudart-dev  {{ cuda_compiler_version }}        # [gpu_variant == "cuda-12"]
+    - libcublas-dev    {{ cuda_compiler_version }}        # [gpu_variant == "cuda-12"]
+    - openblas-devel {{ openblas }}                       # [blas_impl == "openblas"]
+    - mkl-devel {{ mkl }}                                 # [blas_impl == "mkl"]
+    - pthread-stubs 0.3                                   # [linux]
   run:
-    - {{ pin_compatible('cudatoolkit', max_pin='x.x') }}  # [cuda_compiler_version != "None" and not (cuda_compiler_version or "").startswith("12")]
-    - {{ pin_compatible('cuda-version', max_pin='x.x') }} # [cuda_compiler_version != "None"]
-
-    - llvm-openmp                                   # [linux and cuda_compiler_version == "None" and blas_impl == "mkl"]
-    - __cuda >={{ cuda_compiler_version }}          # [cuda_compiler_version != "None" and not (cuda_compiler_version or "").startswith("12")]
+    - {{ pin_compatible('cudatoolkit', max_pin='x.x') }}  # [gpu_variant == "cuda-11"]
+    - {{ pin_compatible('cuda-version', max_pin='x.x') }} # [(gpu_variant == "cuda-11") or (gpu_variant == "cuda-12")]
```

Comment:
we probably want a run requirement on `__OSX >= 12.3` for the metal variant

Reply:
Thanks, addressed

Reply:
In the end I removed the specific sdk setting. This was coming from pytorch feedstock, but doesn't apply to llama, as llama doesn't use the same metal features.

Reply:
Ah right, it's metal not MPS